### PR TITLE
[sil-cse] Add CSE support for open_existential_ref

### PR DESCRIPTION
**Pull request by swiftix**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [x] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by swiftix** [comment]
@swift-ci Please test and merge

**Comment by swiftix** [comment]
@swift-ci Please test and merge

**Comment by swiftix** [comment]
@swift-ci Please test

File: lib/SILOptimizer/Transforms/CSE.cpp [hunk 1]
```diff
@@ -381,14 +390,29 @@ bool llvm::DenseMapInfo<SimpleValue>::isEqual(SimpleValue LHS,
   if (LHS.isSentinel() || RHS.isSentinel())
     return LHSI == RHSI;
 
+  if (isa<OpenExistentialRefInst>(LHSI) && isa<OpenExistentialRefInst>(RHSI)) {
+    // Consider the types of two open_existential_ref instructions to be equal,
+    // if the sets of protocols they conform to are equal.
```
Comment (jrose-apple) [inline lib/SILOptimizer/Transforms/CSE.cpp]: …and their arguments are equal, no?

**Comment by swiftix** [comment]
@swift-ci Pleas test

**Comment by swiftix** [comment]
@swift-ci Please test

**Comment by swiftix** [comment]
The Linux test failure seems to be unrelated, because I see that other PRs had the same issue.

**Pull request merged**